### ISSUE TITLE
MacOS Sierra 10.12.6 .wast files not found / .cpp files should be compile-able to .wast files separately

### ISSUE THREAD

**Issue by comtechnet** [EOSIO/eos#877]
Please make the .cpp files and compile instructions available and PUBLISH the procedure on how to compile them correctly with the correct libraries and resources on MacOS Sierra 10.12.6.

There are multiple folks running into this problem - including me.  I have run both the automatic MacOS scripts and the manual method and STILL have the same issue - .wast files not found - they did NOT get "made" on my MacOS computer, even with PULL request done just a few hours ago.

PLEASE FIX.

**Comment by comtechnet** [comment]
![image](https://user-images.githubusercontent.com/5374172/33642715-2a8e6c86-da09-11e7-8812-acb79c503441.png)

This screenshot says EOSLIB env variable not defined

Why is this NOT in the setup documentation - I'm not a MacOS expert - I don't know where this goes, it sound so fundamental - WHAT AM I MISSING HERE?

**Comment by comtechnet** [comment]
![screen shot 2017-12-05 at 10 17 11 pm](https://user-images.githubusercontent.com/5374172/33642912-28ce8ab0-da0a-11e7-97ef-d0f6a7d9fe0e.png)

I had 2 fatal errors in the chain_test

I'm trying to compile the sample contracts (currency) myself - and I can NOT figure it out.

**Comment by comtechnet** [comment]
![image](https://user-images.githubusercontent.com/5374172/33643503-7dcb994c-da0d-11e7-9b91-b632abaffa8f.png)

So - I've tried to compile the .cpp into a .wast myself and get the error above

Anyone?

**Comment by comtechnet** [comment]
![image](https://user-images.githubusercontent.com/5374172/33643538-bbf496c4-da0d-11e7-8ed7-2ee2b1e3cef6.png)

So - I've tried to compile the .cpp into a .wast file myself for the currency sample app and get the error above.  What am I missing?  Anyone?

**Comment by comtechnet** [comment]
input.cc:2:13: fatal error: 'eoslib/types.h' file not found
   #include <eoslib/types.h>
            ^~~~~~~~~~~~~~~~

I've just REDONE the entire setup after updating XCODE on my iMac MacOS 10.12.6.
Again - the "build" is not compiling the .cpp sample contract files into .wast files.
So - the latest screen shot shows that the manual ./eoscpp (compile) that fails.

So - the bottomline for me: THE STANDARD MacOS 10.12.6 instruction DO NOT WORK!

Does anyone have any advise here - has anyone else had this problem?
Does anyone have suggestions on how to proceed from here?
I'll try the Ubuntu setup next I guess unless someone has some advise/suggestions

**Comment by comtechnet** [comment]
![image](https://user-images.githubusercontent.com/5374172/33749063-dad700a8-db99-11e7-92fb-fcc6c7072915.png)

Here is the entire screenshot of both errors - the original ./eosc set contract currency ../../contracts/currency/currency.wast ../../contracts/currency/currency.abi 

and also the manual compile to try to generate the .wast files from the source .cpp files

**Comment by comtechnet** [comment]
Classic,

My Xcode on my iMac (MacOS Sierra 10.12.6) needed to be updated.  I was using TeamViewer (TV) to remote into a "test" iMac that I reserved to use for this project.  When using TV, and reducing the screen resolution to 1920 x 1080 so that I could see it on my MacBook Pro (from which I was remote-ing in "from"), the remote Dock was not visible with the "Automatically hide and show the dock" setting.  I had "updated" Xcode using the xcode-select --install step and had moved on thru the rest of the install steps.  So for several days after having success with the initial ./eosd setup etc, I had assumed that the pre-req's were completed, because the ./eosd was running configured as my single node-test net, etc.  But ./eosc was giving me errors.

Xcode had partially installed and was beacon-ing from the remote's Dock - which was hidden using TV remotely.  Since the Dock was hidden from view - I did not see it.

Hard lesson learned - once again humbled.  Yet, ironically, as with all these types of incidents, I learned much about EOS and the command line interface, etc.
